tutorials/tutorial-grafana.md

```diff
@@ -0,0 +1,261 @@
+is labeled `time` as Grafana requires, while part 2 is unchanged:
+
+```sql
+SELECT 
+  --1--
+  time_bucket('1 day', pickup_datetime) AS "time",
+  --2--
+  COUNT(*)
+FROM rides 
+```
+
+#### The Grafana __timeFilter command
+
+Grafana time-series panels include a tool that enables the end-user to filter on a given 
+time range. A “time filter,” if you will. Not surprisingly, Grafana has a way to link the 
+user interface construct in a Grafana panel with the query itself. In this case, 
+the `$__timefilter()` function.
+
+In the modified query below, we will enable the Grafana filtering user interface element 
+to set the range for the `pickup_datetime` column in our original New York City taxicab 
+database.
+
+```sql
+SELECT 
+  --1--
+  time_bucket('1 day', pickup_datetime) AS "time",
+  --2--
+  count(*)
+FROM rides
+where $__timeFilter(pickup_datetime)
+```
+
+#### Referencing elements in our query
+
+Finally, we will reference elements of the `SELECT` statement in our query. 
+In our case, we want to group our visualization by the time buckets we’ve selected, 
+and we want to order the results by the time buckets as well. So, our `GROUP BY` 
+and `ORDER BY` statements will refer to the first element of the `SELECT` statement, 
+in this case the `time_bucket` query.
+
+With these changes, this is our final Grafana query:
+
+```sql
+SELECT 
+  --1--
+  time_bucket('1 day', pickup_datetime) AS "time",
+  --2--
+  count(*)
+FROM rides
+where $__timeFilter(pickup_datetime)
+GROUP BY 1
+ORDER BY 1
+```
+
+When we visualize this query in Grafana, we see the following:
+
+<img class="main-content__illustration" src="https://assets.iobeam.com/images/docs/tutorial-grafana/grafana_query_results.png" alt="Visualizing time-series data in Grafana"/>
+
+### Visualize geospatial data stored in TimescaleDB
+
+The NYC Taxi Cab data also contains the location of each ride pickup. In the 
+[Hello, Timescale! Tutorial][hello-timescale], we examined rides that originated 
+near Times Square. Let’s build on that query and 
+**visualize rides whose distance traveled was greater than five miles in Manhattan**.
+
+We can do this in Grafana using the 'Worldmap Panel'. We will start by creating a 
+new panel, selecting 'New Visualization', and selecting the 'Worldmap Panel'.
+
+Once again, we will edit our query directly. In the resulting Query screen, be sure 
+to select your NYC Taxicab Data as the data source. In the 'Format as' dropdown, 
+select 'Table'. Click on 'Edit SQL' and enter the following query in the text window:
+
+```sql
+SELECT time_bucket('1 day', rides.pickup_datetime) AS time,
+       count(rides.trip_distance > 5) AS value, 
+       rides.pickup_latitude AS latitude, 
+       rides.pickup_longitude AS longitude
+FROM rides
+where $__timeFilter(rides.pickup_datetime) AND
+ST_Distance(pickup_geom, ST_Transform(ST_SetSRID(ST_MakePoint(-73.9851,40.7589),4326),2163)) < 2000
```

Comment:
This line needs to be indented and perhaps broken up (just reaches the edge of max length code block window.

<img width="833" alt="Screen Shot 2020-03-04 at 10 00 51 AM" src="https://user-images.githubusercontent.com/12288273/75892331-21015300-5dff-11ea-9631-50823f6702d6.png">